go/engine/engine.go

```diff
@@ -61,16 +61,45 @@ func RunEngine(e Engine, ctx *Context) (err error) {
 	e.G().Log.Debug("+ RunEngine(%s)", e.Name())
 	defer func() { e.G().Log.Debug("- RunEngine(%s) -> %s", e.Name(), libkb.ErrToOk(err)) }()
 
+	if err = delegateUIs(e, ctx); err != nil {
+		return err
+	}
 	if err = check(e, ctx); err != nil {
 		return err
 	}
 	if err = runPrereqs(e, ctx); err != nil {
 		return err
 	}
+
 	err = e.Run(ctx)
 	return err
 }
 
+func delegateUIs(e Engine, ctx *Context) error {
+	if e.G().UIRouter == nil {
+		return nil
+	}
+
+	// currently, only doing this for SecretUI, but in future,
+	// perhaps should iterate over all registered UIs in UIRouter.
```

Comment:
true, this might be the better way to handle the IdentifyUI delegation that I hacked into `service/identify.go`